src/picomsg/src/dummy_oper_db_client/dummy_oper_db_client.cpp

```diff
@@ -109,8 +143,68 @@ bool DummyOperDBObject::do_u64_set(const enum LIBOPER_DB_CLIENT_FIELD_IDS field_
 	return true;
 }
 
-void DummyOperDBObject::log(const int flags) const {
-	std::string log_str = "client_set";
+bool DummyOperDBObject::do_u16_set(const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, uint16_t new_num) {
+	const auto iter = m_fields.find(field_id);
+	if (m_fields.end() != iter) {
+		PICOMSG_LOG(PICOMSG_LOG_INFO, "overwriting field ID %d from %s to %hu",
+					field_id, iter->second.c_str(), new_num);
+	}
+	m_fields.emplace(field_id, std::to_string(new_num));
+	return true;
+}
+
+bool DummyOperDBObject::do_u8_set(const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, uint8_t new_num) {
+	const auto iter = m_fields.find(field_id);
+	if (m_fields.end() != iter) {
+		PICOMSG_LOG(PICOMSG_LOG_INFO, "overwriting field ID %d from %s to %hu",
+					field_id, iter->second.c_str(), new_num);
+	}
+	m_fields.emplace(field_id, std::to_string(new_num));
+	return true;
+}
+
+bool DummyOperDBObject::do_bool_set(const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, bool new_val) {
+	const auto iter = m_fields.find(field_id);
+	if (m_fields.end() != iter) {
+		PICOMSG_LOG(PICOMSG_LOG_INFO, "overwriting field ID %d from %s to %d",
+					field_id, iter->second.c_str(), new_val);
+	}
+	m_fields.emplace(field_id, std::to_string((int)new_val));
+	return true;
+}
+
+bool DummyOperDBObject::do_string_set(const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, const char *new_data) {
+	const auto iter = m_fields.find(field_id);
+	if (m_fields.end() != iter) {
+		PICOMSG_LOG(PICOMSG_LOG_INFO, "overwriting field ID %d from %s to %s",
+					field_id, iter->second.c_str(), new_data);
+	}
+	m_fields.emplace(field_id, new_data);
+	return true;
+}
+
+bool DummyOperDBObject::do_union_set(const enum LIBOPER_DB_CLIENT_FIELD_IDS field_id, struct liboper_union *str_val) {
+	const auto iter = m_fields.find(field_id);
+	if (m_fields.end() != iter) {
+		PICOMSG_LOG(PICOMSG_LOG_INFO, "overwriting field ID %d from %s to %s",
+					field_id, iter->second.c_str(), str_val->str);
```

Comment:
I don't like the fact that the code assumes the union will always contain `str` field. As only liboper has the knowledge of what type the field actually is, I suggest to split the `union_set()` functions to each type we support (e.g. `do_union_str_set()` which will accept a string and the matching field_id of the type it belongs to